### ISSUE TITLE
[CLOSED] Add gradient accumulation as an option

### ISSUE THREAD

**Issue by jeswan** [nyu-mll/jiant-v1-legacy#980]
<a href="https://github.com/pyeres"><img src="https://avatars2.githubusercontent.com/u/6176602?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [pyeres](https://github.com/pyeres)**
_Friday Dec 20, 2019 at 01:55 GMT_
_Originally opened as https://github.com/nyu-mll/jiant/pull/980_

----

This PR adds gradient accumulation as a training option. Gradient accumulation allows us to increase the effective batch size (and use a wider range of GPUs for experiments).

We validated these changes on ReCoRD and BoolQ tasks where we confirmed that we can use gradient accumulation and smaller batch sizes to match the performance of larger batch sizes.

----

_**[pyeres](https://github.com/pyeres)** included the following code: https://github.com/nyu-mll/jiant/pull/980/commits_
